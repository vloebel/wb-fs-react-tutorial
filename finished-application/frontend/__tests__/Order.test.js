import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import Order, { SINGLE_ORDER_QUERY } from '../components/Order';
import { fakeOrder } from '../lib/testUtils';

const order = fakeOrder();
const mocks = [
  {
    request: { query: SINGLE_ORDER_QUERY, variables: { id: 'ord123' } },
    result: { data: { Order: order } },
  },
];

describe('<Order/>', () => {
  it('renders the order', async () => {
    const { container } = render(
      <MockedProvider
        mocks={mocks}
        // Not sure why we need this, but it fixes it. https://github.com/apollographql/apollo-client/issues/7081
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
      >
        <Order id="ord123" />
      </MockedProvider>
    );
    await screen.findByTestId('order');
    expect(container).toMatchSnapshot();
  });
});
