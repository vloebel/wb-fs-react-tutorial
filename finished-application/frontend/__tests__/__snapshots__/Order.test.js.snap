// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order/> renders the order 1`] = `
<div>
  <div
    class="OrderStyles-sc-4oqalm-0 gnIPai"
    data-testid="order"
  >
    <p>
      <span>
        Order ID:
      </span>
      <span>
        ord123
      </span>
    </p>
    <p>
      <span>
        Charge
      </span>
      <span>
        ch_123
      </span>
    </p>
    <p>
      <span>
        Date
      </span>
    </p>
    <p>
      <span>
        Order Total
      </span>
      <span>
        $400
      </span>
    </p>
    <p>
      <span>
        Item Count
      </span>
      <span>
        2
      </span>
    </p>
    <div
      class="items"
    >
      <div
        class="order-item"
      >
        <img />
        <div
          class="item-details"
        >
          <h2>
            soluta non omnis consequatur enim quia autem
          </h2>
          <p>
            Qty: 
            1
          </p>
          <p>
            Each: 
            $42.34
          </p>
          <p>
            SubTotal: 
            $42.34
          </p>
          <p>
            et modi tenetur amet modi reprehenderit omnis
          </p>
        </div>
      </div>
      <div
        class="order-item"
      >
        <img />
        <div
          class="item-details"
        >
          <h2>
            quia sed exercitationem omnis laborum exercitationem est
          </h2>
          <p>
            Qty: 
            1
          </p>
          <p>
            Each: 
            $42.34
          </p>
          <p>
            SubTotal: 
            $42.34
          </p>
          <p>
            sapiente laudantium molestias assumenda quasi adipisci mollitia
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
